lib/clusterable/ht_item.rb

```diff
@@ -44,7 +44,20 @@ def self.with_ocns(ocns)
     end
 
     def self.find(item_id:)
-      from_row(db.where(htid: item_id).first!)
+      from_row(table.where(htid: item_id).first!)
+    end
+
+    # Returns the htitem if it is in the listed cluster, otherwise nil.
+    def self.find_in_cluster(item_id:, ocns:)
+      row = table.select { hf.* }
+        .natural_join(:hf_oclc)
+        .where(value: ocns.map(&:to_s), htid: item_id)
```

Comment:
so this is a `self.find` but with the added constraint that items match the given ocns?
could use a clarifying comment

Reply:
I can also see if there's a common way to extract out the query. It might make more sense to have find take an optional `ocns` parameter.

Reply:
I tried to improve this; might have gone a bit overboard, let me know if it's too hard to follow now..